### ISSUE TITLE
Lift the restriction that limits parallel computing to 5 CPU cores

### ISSUE THREAD

**Issue by 0liver0815** [Roche/crmPack#89]
Currently there is a restriction that limits the number of parallel cores to 5. There may be a reason, but I experienced good performance and no problems using 32 cores in parallel. Therefore, I propose to change the maximum to 32. Or let the user override the maximum with a user specified value (probably the better if the machine is used by more users and one don't want to highjack all cores 😊) 

https://github.com/Roche/crmPack/blob/1874c075f8814a0832578c5affddd4a4985d3783/R/Design-methods.R#L191
https://github.com/Roche/crmPack/blob/1874c075f8814a0832578c5affddd4a4985d3783/R/Design-methods.R#L447

**Comment by danielinteractive** [comment]
@0liver0815 yeah good point. The user can anyway override this "maximum" though by specifying the number of cores manually. 
What we could do is to provide a more clever automated choice here - e.g. we have some code on another package we could port here where the "free" cores are determined. What do you think?

**Comment by 0liver0815** [comment]
@danielinteractive I did not realize that the feature I requested is already implemented. When working with crmpack, I came across the 'problem' that only 5 cores where utilized, but instead of looking into the documentation, I jump to the code and replace 5 by 32 without realizing that nCores is already a parameter of the function. Therefore, no change necessary.
However, the additional feature that you are mentioning sounds very cool to me. I find this a very reasonable approach that only the free capacity is utilized. In case that the implementation is not to time-consuming for you, I would like to see such an automatic selection.

**Comment by 0liver0815** [comment]
I investigated this further and can confirm that the number of cores is limited to 5, regardless what is specified in `nCores`. The reason is that the following parameter is handed over to the method `getResultList`, e.g. here

https://github.com/Roche/crmPack/blob/1874c075f8814a0832578c5affddd4a4985d3783/R/Design-methods.R#L384

and `getResultList` determine the number of cores in the following way:

https://github.com/Roche/crmPack/blob/1874c075f8814a0832578c5affddd4a4985d3783/R/Design-methods.R#L104-L106

This will result in a maximum of 5 cores.

@danielinteractive will make a proposal for the Design object in a PR.

**Comment by danielinteractive** [comment]
cool thanks @0liver0815 ! good that you found where this 5 is - I really don't remember that anymore. PR will be nice, otherwise when @wwojciech is getting there with the clean up he can also keep this in mind and fix then
